pkg/data_cache/test/src/main.rs

```diff
@@ -0,0 +1,108 @@
+//! Sample client for validating access to cached datasets in the Arrow-based caching system.
+//!
+//! This client demonstrates how to connect to the head node and stream cached data
+//! from worker nodes using Apache Arrow Flight protocol. It serves as a reference
+//! implementation for accessing and consuming distributed cached datasets.
+
+use arrow_flight::decode::FlightRecordBatchStream;
+use futures::stream::StreamExt;
+use arrow_flight::flight_service_client::FlightServiceClient;
+use tonic::transport::Channel;
+use arrow_flight::FlightDescriptor;
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+use tracing::{info, error};
+use tracing_subscriber;
+use clap::Parser;
+
+/// Index pair structure for dataset partitioning
+#[derive(Serialize, Deserialize, Debug)]
+struct IndexPair {
+    start: u64,
+    end: u64,
+}
```

Comment:
Can we use IndexPair from the config?
```suggestion
```

Reply:
Actually index pair is not used in test after the latest refactor